### DISCUSSION TITLE
how efficient the message migration of ActiveMQ Operator is？

### DISCUSSION THREAD

**Discussion by Turing-Lee666** [arkmq-org/arkmq-org-broker-operator#1065] [Q&A]
Hi: 
I want to know how efficient the message migration of ActiveMQ Operator is in the scenario of reducing the number of replicas. Can you introduce the implementation mechanism of message migration? And will it have an impact on customer business when scaling down?

**Comment by brusdev** [comment]
@Turing-Lee666 the message migration process is the following: when the deployment size is scaled down, the scaledown controller starts a drainer Pod. The drainer Pod connects to one of the other live broker Pods in the cluster and migrates messages to that live broker Pod. The message migration time depends on the number of the messages. The main impact of this process is that during the procedure the migrating messages are not available.

**Reply by Turing-Lee666** [reply]
Thanks
